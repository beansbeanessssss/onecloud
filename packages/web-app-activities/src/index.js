import translations from '../l10n/translations'
import Activities from './views/Activities.vue'
// just a dummy function to trick gettext tools
function $gettext(msg) {
  return msg
}

const appInfo = {
  name: $gettext('Activities'),
  id: 'activities',
  icon: 'activities',
  isFileEditor: false
}

const routes = [
  {
    path: '/',
    name: 'activities-all',
    component: Activities,
    meta: {
      title: $gettext('Users')
    }
  }
]

const navItems = []

export default {
  appInfo,
  routes,
  translations,
  navItems
}
