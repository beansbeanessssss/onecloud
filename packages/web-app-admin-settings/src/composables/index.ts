export * from './actions'
export * from './usePagination'
