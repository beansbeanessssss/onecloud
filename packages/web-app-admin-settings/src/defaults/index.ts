export const supportedLogoMimeTypes = ['image/gif', 'image/jpeg', 'image/png']
export const perPageDefault = '50'
export const perPageStoragePrefix = 'admin-settings'
export const paginationOptions = ['20', '50', '100', '250']
