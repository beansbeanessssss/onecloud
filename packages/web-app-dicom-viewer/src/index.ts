import translations from '../l10n/translations.json'
import { AppWrapperRoute } from 'web-pkg/src/components/AppTemplates/AppWrapperRoute'
import DicomViewer from './DicomViewer.vue'

// just a dummy function to trick gettext tools
function $gettext(msg) {
  return msg
}

const routes = [
  {
    path: '/:driveAliasAndItem(.*)?',
    component: AppWrapperRoute(DicomViewer, {
      applicationId: 'dicom-viewer',
      urlForResourceOptions: {
        //disposition: 'inline'
      }
    }),
    name: 'dicom-viewer',
    meta: {
      authContext: 'hybrid',
      title: $gettext('DICOM Viewer'),
      patchCleanPath: true
    }
  }
]

const appInfo = {
  name: $gettext('DICOM Viewer'),
  id: 'dicom-viewer',
  icon: 'resource-type-medical',
  iconFillType: 'fill',
  iconColor: 'var(--oc-color-icon-medical)',
  extensions: [
    {
      extension: 'dcm',
      routeName: 'dicom-viewer'
    }
  ]
}

export default {
  appInfo,
  routes,
  translations
}
