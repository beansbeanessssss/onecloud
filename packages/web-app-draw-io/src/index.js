import translations from '../l10n/translations'
import App from './App.vue'

const routes = [
  {
    name: 'edit',
    path: '/edit/:filePath',
    components: {
      fullscreen: App
    },
    meta: { hideHeadbar: true }
  }
]

const appInfo = {
  name: 'Draw.io',
  id: 'draw-io',
  icon: 'grid_on',
  extensions: [
    {
      extension: 'drawio',
      newTab: true,
      routeName: 'draw-io-edit',
      newFileMenu: {
        menuTitle($gettext) {
          return $gettext('New draw.io documentâ€¦')
        }
      },
      routes: [
        'files-spaces-storage',
        'files-common-favorites',
        'files-shares-with-others',
        'files-shares-with-me',
        'files-public-files'
      ]
    },
    {
      extension: 'vsdx',
      newTab: true,
      routeName: 'draw-io-edit',
      routes: [
        'files-spaces-storage',
        'files-common-favorites',
        'files-shares-with-others',
        'files-shares-with-me',
        'files-public-files'
      ]
    }
  ]
}

export default {
  appInfo,
  routes,
  translations
}
