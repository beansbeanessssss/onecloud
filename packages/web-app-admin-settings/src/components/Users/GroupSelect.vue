<template>
  <div id="user-group-select-form">
    <oc-select
      :model-value="selectedOption"
      class="oc-mb-s"
      :multiple="true"
      :options="groupOptions"
      option-label="displayName"
      :label="$gettext('Groups')"
      :fix-message-line="true"
      @update:model-value="onUpdate"
    >
      <template #selected-option="{ displayName, id }">
        <span class="oc-flex oc-flex-center">
          <avatar-image
            class="oc-flex oc-align-self-center oc-mr-s"
            :width="16.8"
            :userid="id"
            :user-name="displayName"
          />
          <span>{{ displayName }}</span>
        </span>
      </template>
      <template #option="{ displayName, id }">
        <div class="oc-flex">
          <span class="oc-flex oc-flex-center">
            <avatar-image
              class="oc-flex oc-align-self-center oc-mr-s"
              :width="16.8"
              :userid="id"
              :user-name="displayName"
            />
            <span>{{ displayName }}</span>
          </span>
        </div>
      </template>
    </oc-select>
  </div>
</template>
<script lang="ts">
import { defineComponent, PropType, ref, unref, watch } from 'vue'
import { computed } from 'vue'
import { Group } from 'web-client/src/generated'

export default defineComponent({
  name: 'GroupSelect',
  props: {
    selectedGroups: {
      type: Array as PropType<Group[]>,
      required: true
    },
    groupOptions: {
      type: Array as PropType<Group[]>,
      required: true
    }
  },
  emits: ['selectedOptionChange'],
  setup(props, { emit }) {
    const selectedOption = ref(props.selectedGroups)
    const onUpdate = (event) => {
      selectedOption.value = event
      emit('selectedOptionChange', unref(selectedOption))
    }

    const currentGroups = computed(() => props.selectedGroups)
    watch(currentGroups, () => {
      selectedOption.value = props.selectedGroups
    })

    return { selectedOption, onUpdate }
  }
})
</script>
