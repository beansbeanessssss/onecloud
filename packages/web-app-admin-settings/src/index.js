import translations from '../l10n/translations'
import Users from './views/Users.vue'
import Groups from './views/Groups.vue'
import Spaces from './views/Spaces.vue'
// just a dummy function to trick gettext tools
function $gettext(msg) {
  return msg
}

const appInfo = {
  name: $gettext('Administration Settings'),
  id: 'admin-settings',
  icon: 'settings-4',
  isFileEditor: false
}

const routes = [
  {
    path: '/',
    redirect: () => {
      const permissionManager = window.Vue.$permissionManager
      if (permissionManager.hasUserManagement()) {
        return { name: 'admin-settings-users' }
      }
      return { name: 'admin-settings-spaces' }
    }
  },
  {
    path: '/users',
    name: 'admin-settings-users',
    component: Users,
    meta: {
      authContext: 'user',
      title: $gettext('Users')
    }
  },
  {
    path: '/groups',
    name: 'admin-settings-groups',
    component: Groups,
    meta: {
      authContext: 'user',
      title: $gettext('Groups')
    }
  },
  {
    path: '/spaces',
    name: 'admin-settings-spaces',
    component: Spaces,
    meta: {
      authContext: 'user',
      title: $gettext('Spaces')
    }
  }
]

const navItems = [
  {
    name: $gettext('Users'),
    icon: 'user',
    route: {
      path: `/${appInfo.id}/users?`
    },
    enabled: () => {
      const permissionManager = window.Vue.$permissionManager
      return permissionManager.hasUserManagement()
    }
  },
  {
    name: $gettext('Groups'),
    icon: 'group-2',
    route: {
      path: `/${appInfo.id}/groups?`
    },
    enabled: () => {
      const permissionManager = window.Vue.$permissionManager
      return permissionManager.hasUserManagement()
    }
  },
  {
    name: $gettext('Spaces'),
    icon: 'layout-grid',
    route: {
      path: `/${appInfo.id}/spaces?`
    },
    enabled: () => {
      const permissionManager = window.Vue.$permissionManager
      return permissionManager.hasSpaceManagement()
    }
  }
]

export default {
  appInfo,
  routes,
  translations,
  navItems
}
