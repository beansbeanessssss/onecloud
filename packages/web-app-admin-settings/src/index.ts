import translations from '../l10n/translations.json'
import General from './views/General.vue'
import Users from './views/Users.vue'
import Groups from './views/Groups.vue'
import Spaces from './views/Spaces.vue'
// just a dummy function to trick gettext tools
function $gettext(msg) {
  return msg
}

const appInfo = {
  name: $gettext('Administration Settings'),
  id: 'admin-settings',
  icon: 'settings-4',
  isFileEditor: false
}

// FIXME: a better way to access this is needed
const permissionManager = () => (window as any).__$permissionManager

const routes = [
  {
    path: '/',
    redirect: () => {
      if (permissionManager().hasSystemManagement()) {
        return { name: 'admin-settings-general' }
      }
      return { name: 'admin-settings-spaces' }
    }
  },
  {
    path: '/general',
    name: 'admin-settings-general',
    component: General,
    meta: {
      authContext: 'user',
      title: $gettext('General')
    }
  },
  {
    path: '/users',
    name: 'admin-settings-users',
    component: Users,
    meta: {
      authContext: 'user',
      title: $gettext('Users')
    }
  },
  {
    path: '/groups',
    name: 'admin-settings-groups',
    component: Groups,
    meta: {
      authContext: 'user',
      title: $gettext('Groups')
    }
  },
  {
    path: '/spaces',
    name: 'admin-settings-spaces',
    component: Spaces,
    meta: {
      authContext: 'user',
      title: $gettext('Spaces')
    }
  }
]

const navItems = [
  {
    name: $gettext('General'),
    icon: 'settings-4',
    route: {
      path: `/${appInfo.id}/general?`
    },
    enabled: () => {
      return permissionManager().hasSystemManagement()
    }
  },
  {
    name: $gettext('Users'),
    icon: 'user',
    route: {
      path: `/${appInfo.id}/users?`
    },
    enabled: () => {
      return permissionManager().hasUserManagement()
    }
  },
  {
    name: $gettext('Groups'),
    icon: 'group-2',
    route: {
      path: `/${appInfo.id}/groups?`
    },
    enabled: () => {
      return permissionManager().hasUserManagement()
    }
  },
  {
    name: $gettext('Spaces'),
    icon: 'layout-grid',
    route: {
      path: `/${appInfo.id}/spaces?`
    },
    enabled: () => {
      return permissionManager().hasSpaceManagement()
    }
  }
]

export default {
  appInfo,
  routes,
  translations,
  navItems
}
