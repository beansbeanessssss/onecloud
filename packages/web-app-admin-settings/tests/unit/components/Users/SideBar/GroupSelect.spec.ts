import GroupSelect from '../../../../../src/components/Users/SideBar/GroupSelect.vue'
import { defaultPlugins, shallowMount } from 'web-test-helpers'
import { mock } from 'jest-mock-extended'
import { Group } from 'web-client/src/generated'

const groupMock = mock<Group>({ id: '1' })

describe('GroupSelect', () => {
  it('renders the select input', () => {
    const { wrapper } = getWrapper()
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('emits "selectedOptionChange" on update', () => {
    const group = mock<Group>({ id: '2' })
    const { wrapper } = getWrapper()
    wrapper.vm.onUpdate(group)
    expect(wrapper.emitted().selectedOptionChange).toBeTruthy()
    expect(wrapper.vm.selectedOption).toEqual(group)
  })
})

function getWrapper() {
  return {
    wrapper: shallowMount(GroupSelect, {
      props: {
        selectedGroups: [groupMock],
        availableGroups: [groupMock]
      },
      global: {
        plugins: [...defaultPlugins()]
      }
    })
  }
}
