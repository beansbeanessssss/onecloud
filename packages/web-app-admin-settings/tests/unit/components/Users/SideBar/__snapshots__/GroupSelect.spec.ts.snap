// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupSelect renders the select input 1`] = `
<div id="user-group-select-form">
  <oc-select-stub class="oc-mb-s" clearable="false" disabled="false" filter="[Function]" fix-message-line="true" id="oc-select-1" label="Groups" loading="false" model-value="undefined" multiple="" optionlabel="displayName" options="undefined" searchable="true"></oc-select-stub>
</div>
`;
